nbgitpuller/handlers.py

```diff
@@ -73,8 +132,21 @@ def get(self):
             self.set_header('content-type', 'text/event-stream')
             self.set_header('cache-control', 'no-cache')
 
-            gp = GitPuller(repo, repo_dir, branch=branch, depth=depth, parent=self.settings['nbapp'])
+            # if content_provider is specified then we are dealing with compressed
+            # archive and not a git repo
+            if content_provider is not None:
+                pm = self.setup_plugins(content_provider)
+                req_args = {k: v[0].decode() for k, v in self.request.arguments.items()}
+                download_q = Queue()
+                req_args["progress_func"] = lambda: self._wait_for_sync_progress_queue(download_q)
```

Comment:
I asked Min RK on Gitter for help to understand if there was a difference between using a `lambda: my_function(my_arg)` vs `partial(my_function, my_arg)` and learned there was a slight difference, but not a large one. I understand it as it wouldn't make a difference here, but it seems like the more robust and less error prone option to use a partial, I'd like to use it here anyhow.

We need to also do `from functools import partial`

```suggestion
                req_args["progress_func"] = partial(self._wait_for_sync_progress_queue, download_q)
```

MinRK made a gist about the difference here: https://gist.github.com/minrk/597c240bb9a120dd004b0de6cf1d5a9d, and a key point about the difference was:

> the difference is when you create the partial, you are assigning an attribute to an object (partial.args = (0,)). Wheras the lambda only has a reference to the calling scope which is a long-lived mutable object.
(a dict, ultimately)

For the full conversation, see the jupyterhub/random chat on gitter, 2021-11-18